class.jetpack-gutenberg.php

```diff
@@ -155,109 +164,31 @@ static function load( $response = null, $handler = null, $request = null ) {
 		 *
 		 * @param boolean
 		 */
-		if ( apply_filters( 'jetpack_load_beta_blocks', $is_availability_endpoint_beta ) ) {
+		if ( apply_filters( 'jetpack_load_beta_blocks', false ) ) {
 			Jetpack_Constants::set_constant( 'JETPACK_BETA_BLOCKS', true );
 		}
 
 		/**
-		 * Filter the list of block editor blocks that are available through jetpack.
-		 *
-		 * This filter is populated by Jetpack_Gutenberg::jetpack_set_available_blocks
+		 * Filter the whitelist of block editor extensions that are available through Jetpack.
 		 *
 		 * @since 6.8.0
 		 *
 		 * @param array
 		 */
-		self::$blocks = apply_filters( 'jetpack_set_available_blocks', self::$default_blocks );
-
-		/**
-		 * Filter the list of block editor plugins that are available through jetpack.
-		 *
-		 * This filter is populated by Jetpack_Gutenberg::jetpack_set_available_blocks
-		 *
-		 * @since 6.9.0
-		 *
-		 * @param array
-		 */
-		self::$plugins = apply_filters( 'jetpack_set_available_plugins', self::$default_plugins );
-		self::set_blocks_availability();
-		self::set_plugins_availability();
-		self::register_blocks();
-		return $response;
-	}
-
-	static function is_registered( $slug ) {
-		return isset( self::$registered[ $slug ] );
+		self::$extensions = apply_filters( 'jetpack_set_available_blocks', self::get_jetpack_gutenberg_extensions_whitelist() );
```

Comment:
Are we still calling this filter at the same runtime as before?
We it used to be called at action `init` priority 90. This would make it more usable in plugins.

Reply:
tl;dr: It's still possible to add the filter prior to initialization.

So the previous behavior was:

- Define an array of [`$default_blocks`](https://github.com/Automattic/jetpack/blob/91d89bc2033bf42fd6993ac712878866f46d4a5b/class.jetpack-gutenberg.php#L48-L66).
- In `init()`, [apply](https://github.com/Automattic/jetpack/blob/91d89bc2033bf42fd6993ac712878866f46d4a5b/class.jetpack-gutenberg.php#L182) the `jetpack_set_available_blocks` filter to those.
- Define a static method _also_ called [`jetpack_set_available_blocks`](https://github.com/Automattic/jetpack/blob/91d89bc2033bf42fd6993ac712878866f46d4a5b/class.jetpack-gutenberg.php#L301-L323).
- That method is added to the filter of the same name _outside_  of the class, in `class.jetpack.php`: https://github.com/Automattic/jetpack/blob/91d89bc2033bf42fd6993ac712878866f46d4a5b/class.jetpack.php#L543

This has been very confusing to me, and I'd say that it arguably breaks encapsulation: if we're both applying the filter _and_ providing a default method to use it with that even bears the same name, it's very surprising to see that filter added outside the class.

Furthermore, `$default_blocks` has long been confusing people, see #11075.

So removing `$default_blocks` gives us the opportunity to use the return value of the `jetpack_set_available_blocks` _method_ (which I'm renaming, to tell it apart from the filter) as the 'default' in `apply_filters`, while still giving outside stakeholders the chance to add their own filters on top of that. This is e.g. still used in the unit test.

There are conceptual reasons why I believe we shouldn't modify the whitelist of allowed blocks after initialization. In OOP parlance, this is our invariant: If we change it after the fact, what does that mean for blocks that were registered before that change? It's easy to create a mess of badly defined states here. I'll say that ideally, I think this class should become even more encapsulated -- i.e. turn a lot of `static` stuff into regular instance methods and variables, turn the `init` into a constructor. (We would still leave it to consumers to invoke that constructor by creating a new object, and could pass data like the whitelist as an arg to the constructor.) This is the kind of OOP tooling that allows to preserve those invariants (and hence keep state sane). I didn't make that change because it would've inflated the size of this PR even more.